pkg/logger/logger.go

```diff
@@ -1,26 +1,59 @@
+/*
+Logging library for the troubleshoot framework.
+
+Logging levels
+TODO: Document me here => https://github.com/replicatedhq/troubleshoot/issues/1031
+
+0: also the same as not using V() log progress related information within the framework. Logs within each component (collector/analyzers/etc) should not use this level.
+
+1: High level logs within each component (collector/analyzers/etc) should use this level. A log such as "Ceph collector connected to the cluster" belongs here.
+
+2: Everything else goes here. If you do not know which level to use, use this level.
+
+The best approach is to always use V(2) then after testing your code as a whole, you can elevate the log level of the messages you find useful to V(1) or V(0).
+
+Do not log errors in functions that return an error. Instead, return the error and let the caller log it.
+*/
 package logger
 
 import (
-	"log"
-	"os"
-)
+	"flag"
 
-var (
-	logger *log.Logger
-	quiet  = false
+	"github.com/go-logr/logr"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	"k8s.io/klog/v2"
 )
 
-func init() {
-	logger = log.New(os.Stderr, "", log.LstdFlags)
+// InitKlogFlags initializes klog flags and adds them to the cobra command.
+func InitKlogFlags(cmd *cobra.Command) {
+	// Initialize klog flags
+	klogFlags := flag.NewFlagSet("klog", flag.ExitOnError)
+	klog.InitFlags(klogFlags)
+
+	klogFlags.VisitAll(func(f *flag.Flag) {
+		// Just the flags we want to expose in our CLI
+		if f.Name == "v" {
+			// If we ever want to expose the klog flags that have underscores ("_") in them
+			// we need to replace them with hyphens ("-") in the flag name using
+			// pflag.NormalizedName(strings.ReplaceAll(name, "_", "-")). Check how kubectl does it
+			cmd.Flags().AddGoFlag(f)
+		}
+	})
 }
 
-func SetQuiet(s bool) {
-	quiet = s
+// SetupLogger sets up klog logger based on viper configuration.
+func SetupLogger(v *viper.Viper) {
+	quiet := v.GetBool("debug") || v.IsSet("v")
+	SetQuiet(!quiet)
 }
 
-func Printf(format string, args ...interface{}) {
+// SetQuiet enables or disables klog logger.
+func SetQuiet(quiet bool) {
 	if quiet {
-		return
+		klog.SetLogger(logr.Discard())
+	} else {
+		// Restore the default logger
+		klog.ClearLogger()
```

Comment:
do we need to add `klog.ClearLogger()`? I saw SetupLogger has been used in all the root go without any SetLogger before it if we are not use debug or v flag.

If we are going to use it not just in initial stage function, 
then I saw one notice in the klog docs about [ClearLogger](https://pkg.go.dev/k8s.io/klog/v2#ClearLogger) 
```
Modifying the logger is not thread-safe and should be done while no other goroutines invoke log calls, usually during program initialization.
```

Reply:
Good callout on thread safety. I'll update it.

> do we need to add klog.ClearLogger()? I saw SetupLogger has been used in all the root.go without any SetLogger before it if we are not use debug or v flag.

Think of `klog.ClearLogger()` as a way to restore klog to its defaults. I think this function should have been `klog.RestoreDefaultLogger()` or something similar.